### DISCUSSION TITLE
Support full sitemap spec in sitemap metadata route

### DISCUSSION THREAD

**Discussion by LuudJanssen** [vercel/next.js#48479] [Ideas]
### Goals

Let the sitemap metadata route allow adding all properties described by the specification.

### Non-Goals

It is mentioned in the docs that multi sitemap files are WIP and will be added later, so I consider this out-of-scope for now, making this feature request low hanging fruit.

### Background

Currently, only the `loc` and `lastmod` properties of a `url` in the generated sitemap, generated by a sitemap metadata route (`/app/sitemap.ts`) are supported. However, [the spec](https://www.sitemaps.org/protocol.html#xmlTagDefinitions) also defines `changefreq` and `priority` as possible attributes.

When using the sitemap metadata routes in a commercial project I was surprised to find these attributes not supported.

### Proposal

Add the `priority` and `changefreq` attributes to the sitemap entries for the sitemap metadata routes.

A possible additional feature could be the automatic escaping of URL's in the generated XML, also described by [the spec](https://www.sitemaps.org/protocol.html#escaping).

**Comment by LuudJanssen** [comment]
On a slightly more critical note, I'd like to ask why these properties weren't added in the first place. It feels to me like the feature shouldn't have been released when support is still very limited. Making the sitemaps fully spec compliant also doesn't feel like a big thing to add, making me feel like the dev team just forgot to read the specification.

Don't get me wrong, loving the feature, just wanted to mention my disappointment due to the lack of attention to detail. Of course, given there's no good reason to omit these properties from the (first) release.

**Comment by LuudJanssen** [comment]
I [opened a PR](https://github.com/vercel/next.js/pull/48484) for the addition of the attributes. Since it was a small change, I didn't wait for confirmation from the Next.js team to make it easy for them to just accept the PR instead of doing the whole discussion.

I didn't add the URL escaping, because it could possibly result in backwards incompatible changes.

**Comment by stephtr** [comment]
This is actually not the only missing part. There should for example also be a way to define alternative versions, see [Google Search Central docs](https://developers.google.com/search/docs/specialty/international/localized-versions#sitemap) on that, although that's not standard.

**Comment by iljamulders** [comment]
Also missing:
https://developers.google.com/search/docs/specialty/international/localized-versions#sitemap

**Reply by stephtr** [reply]
Thanks, that's the link I actually wanted to paste.

**Comment by ozgur-shape** [comment]
I can't believe those pieces are missing. Sitemaps are a critical part of a website. This will effect us very badly. 😞

**Comment by hungcrush** [comment]
how's it going on?.....

**Comment by sjaanus** [comment]
It is now supported from v13.4.5
